kubernetes/linux/main.sh

```diff
@@ -334,32 +334,35 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
+            registry="mcr.microsoft.com"
+            if  [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
+                  registry=$MCR_URL
```

Comment:
This requires change in AKS side.

Reply:
Not able to create a PR there because in the process of getting permissions. Will merge this PR after that is done

Reply:
Created a PR on aks-rp